testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -152,13 +152,53 @@
 # Check a checkbox of the given id
 #
 When(/^I check "([^"]*)"$/) do |identifier|
-  check(identifier)
-  raise ScriptError, "Checkbox #{identifier} not checked." unless has_checked_field?(identifier)
+  begin
+    contenedor = find('div', text: identifier, exact_text: true)
+    checkbox_element = contenedor.find('input[type="checkbox"]')
+    checkbox_id = checkbox_element[:id]
+  rescue StandardError => e
+    log "Error message: #{e.message}"
+    # Error not raised, it is just to try to uncheck the checkbox if it is not structured as before
+    checkbox_id = nil
+  end
+  # Given that this is an action instead of an assert makes sense not performing it if the checkbox is not previosusly unchecked
+  if has_unchecked_field?(identifier)
+    if checkbox_id.nil?
+      check(identifier)
+    else
+      check(identifier, id: checkbox_id)
+    end
+    sleep(0.5)
+    raise ScriptError, "Checkbox #{identifier} not checked." if has_unchecked_field?(identifier)
+  else
+    log "Checkbox:#{identifier} is already checked"
```

Comment:
What about just that?

```suggestion
    checkbox = find(:xpath, "//div[. = '#{text}']//input[@type='checkbox']")
    checkbox.set(true)
  rescue StandardError => e
    log "Error message: #{e.message}"
```

Reply:
Done